### PR TITLE
Chore: Create a basic test workflow

### LINKED ISSUE #241: [chore] Add unit tests, or remove test from workflow?
Currently there is an `npm run test` call in the current workflow, but there are no tests. There is `jest` and `ts-jest` dependencies so maybe it might be time to look back into adding some basic unit tests? 

Or removing these tests could provide some cleanup.

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #241

I created a starting test suite filled with a test for the millisecond calculation utility. I also decided to make a tiny adjustment to the calculate millisecond utility by making all units be converted to lower case and completing the typescript typing for it.

### CODE REVIEW

File: .eslintignore
Comment (naomi-lgbt) [inline .eslintignore:1]: This seems to be from a different repo and not necessary here.

File: src/utils/calculateMilliseconds.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@
  * @param {string} unit The unit of time to convert to milliseconds.
  * @returns {number} The number of milliseconds.
  */
-export const calculateMilliseconds = (value: number, unit: string) => {
-  switch (unit) {
+export const calculateMilliseconds = (value: number, unit: string): number => {
+  switch (unit.toLowerCase()) {
```
Comment (naomi-lgbt) [inline src/utils/calculateMilliseconds.ts:10]:
```suggestion
export const calculateMilliseconds = (value: number, unit: "seconds" | "minutes" | "hours" | "days" | "weeks"): number => {
  switch (unit) {
```

We could actually get more strict with the typing and avoid potential issues.

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
   "devDependencies": {
     "@types/common-tags": "1.8.1",
     "@types/diff": "5.0.3",
+    "@types/jest": "^29.5.0",
```
Comment (naomi-lgbt) [inline package.json]:
```suggestion
    "@types/jest": "29.5.0",
```

Will also need to run `npm i` after committing this to update the package-lock.json

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: .eslintignore
Comment (a2937) [reply .eslintignore:1]: That's correct and completely my fault. I'll remove it.

File: src/utils/calculateMilliseconds.ts (hunk 1 above)
Comment (a2937) [reply src/utils/calculateMilliseconds.ts:10]: Thank you.

File: package.json (hunk 2 above)
Comment (a2937) [reply package.json]: Done.

**Comment by a2937** [comment]
Looks like I need to move the test folder over

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
We're almost there.

The `tsconfig.json` will need to have an `exclude` property added, set to an array that contains `./test`.

[Example](https://github.com/BeccaLyria/discord-bot/blob/main/tsconfig.json).

**Comment by a2937** [comment]
:/ I just wanted to make a simple pull request.

File: src/utils/calculateMilliseconds.ts (hunk 1 above)
Comment (a2937) [reply src/utils/calculateMilliseconds.ts:10]: However we can't actually do that as I just found out. Because the mute command will fail verification.

**Review by naomi-lgbt — APPROVED** [review]
Happy to land this as is. I'll circle back and re-implement the type def + write a type guard.

**Pull request merged**